src/main/java/edu/umich/ctools/sectionsUtilityTool/Friend.java

```diff
@@ -0,0 +1,317 @@
+package edu.umich.ctools.sectionsUtilityTool;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.lang.reflect.*;
+import java.io.IOException;
+import java.io.FileInputStream;
+import java.io.File;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.Principal;
+import java.security.PrivateKey;
+import java.security.NoSuchAlgorithmException;
+import java.security.KeyStoreException;
+import java.security.KeyManagementException;
+import java.security.UnrecoverableKeyException;
+import java.security.interfaces.RSAPrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.cert.X509Certificate;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.security.cert.CertificateException;
+import java.security.SecureRandom;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.KeyFactory;
+
+import javax.net.ssl.*;
+import javax.net.SocketFactory;
+import javax.mail.*;
+import javax.mail.internet.*;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.xmlrpc.client.XmlRpcClient;
+import org.apache.xmlrpc.client.XmlRpcClientConfigImpl;
+import org.apache.xmlrpc.client.XmlRpcSun15HttpTransportFactory;
+
+import edu.umich.its.lti.utils.PropertiesUtilities;
+
+//TODO - Email Templates
+
+public class Friend 
+{
+
+	private static Log M_log = LogFactory.getLog(Friend.class);
+
+	//Currently it uses xmlrpc for communication
+	public static XmlRpcClient Xclient;
+
+	protected static String friendUrl = null;
+	protected static String contactEmail = null;
+	protected static String emailMessage = null;
+	protected static String referrerUrl = null;
+	protected static String ksFileName = null;
+	protected static String ksPwd = null;
+	protected static String friendEmailFile = null;
+	protected static String requesterEmailFile = null;
+
+	public final String KEYSTORETYPE_PKCS12 = "pkcs12";
+
+	public final String TRUSTSTORETYPE_JKS = "jks";
+
+	private static boolean sslInitialized = false;
+	
+	protected static final String FRIEND_PROPERTY_FILE_PATH_SECURE = "sectionsToolFriendPropsPathSecure";
+	
+	protected static Properties appExtSecurePropertiesFile=null;
+	
+
+	public void init()
+	{
+		M_log.info(this +".init()");
+		
+		String propertiesFilePathSecure = System.getProperty(FRIEND_PROPERTY_FILE_PATH_SECURE);
+		M_log.info("props: " + propertiesFilePathSecure);
+		if (!propertiesFilePathSecure.isEmpty()) {
```

Comment:
How about error message and return / exit if the file path is empty?  Can it run correctly without the properties?